### PR TITLE
Add strategy tests.

### PR DESCRIPTION
**Pull request by andreif**
✅ Ready for review.

### CODE REVIEW

**Comment by andreif** [comment]
Not ready to merge. I think I made incorrect render calls.

**Comment by andreif** [comment]
Yeah I did, now fixed correctly and ready for review.

File: social_django/compat.py [hunk 1]
```diff
@@ -18,3 +18,11 @@ def get_rel_model(field):
...
         user_model = models.get_model(app_label, model_name)
     return user_model
+
+
+def get_request_port(request):
+    if django.VERSION >= (1, 9):
+        return request.get_port()
+
+    host_parts = request.get_host().partition(':')
+    return host_parts[2] or request.META['SERVER_PORT']
```
Comment (omab) [inline social_django/compat.py]: Should be `host_parts[1]`?

File: social_django/strategy.py [hunk 2]
```diff
@@ -142,12 +133,8 @@ def build_absolute_uri(self, path=None):
             return path
 
     def random_string(self, length=12, chars=BaseStrategy.ALLOWED_CHARS):
-        try:
-            from django.utils.crypto import get_random_string
-        except ImportError:  # django < 1.4
-            return super(DjangoStrategy, self).random_string(length, chars)
-        else:
-            return get_random_string(length, chars)
+        from django.utils.crypto import get_random_string
```
Comment (omab) [inline social_django/strategy.py]: I know it's not actually your code, but could you move this import to the top with the rest of the django ones?

**Comment by omab** [comment]
@andreif, great work, I've left a couple very small comments.

File: social_django/compat.py (hunk 1 above)
Comment (andreif) [reply social_django/compat.py]: `'1,2'.partition(',') == ['1', ',', '2']`

File: social_django/strategy.py (hunk 2 above)
Comment (andreif) [reply social_django/strategy.py]: sure thing 🙂

File: social_django/compat.py (hunk 1 above)
Comment (omab) [reply social_django/compat.py]: Gotcha, it's a partition, not an split. Thanks.

**Comment by omab** [comment]
Thanks!

**Pull request merged**